### DISCUSSION TITLE
No CMake_Fortran_COMPILER could be found.

### DISCUSSION THREAD

**Discussion by jas220** [ImperialCollegeLondon/sharpy#135] [Q&A] [answered]
Hi,
I was following the SHARPy Installation Guide and on step 4 of the quick installation, where I had to prepare UVLM and xbeam for compilation using gfortran and g++, I ran the “cmake ..” command and ran into this error (attached). I am currently running a macOS Big Sur Version 11.2.3. I’ve tried downloading a Fortran compiler, but I keep encountering the same issue.
Thank you for your help in advance!

![CMAKE_Fortran_Error](https://user-images.githubusercontent.com/82110485/113912108-af01f700-97d2-11eb-8562-4bd091150a28.jpg)

**Comment by ngoiz** [comment, marked as answer]
Hello,

I might be wrong on this but let's try a few things. I assume you have `gfortran` installed. If you run on the shell `which gfortran` what do you get?

The issue that CMAKE is showing is that it cannot find the compiler, you can manually specify where to look for it by running:
`
FC=/path/to/gfortran CXX=path/to/gcc cmake ..
`

Give this a go see if it works. If not we'll try and figure something out

**Reply by jas220** [reply]
Hello, 

Sorry for the delayed response. I tried doing it all from the beginning, even downloading and installing the gfortran compiler again, and I encountered the same error. I, then, tried what you suggested and got the following result.

<img width="641" alt="Screen Shot 2021-04-20 at 9 35 01 PM" src="https://user-images.githubusercontent.com/82110485/115461497-6827ee80-a221-11eb-9ed7-13ac0010230c.png">

**Reply by fonsocarre** [reply]
Hi, when you do
`FC=/path/to/fortran CXX=/path/to/gcc cmake ..`, substitute /path/to/fortran with whatever comes out of `which gfortran` (/usr/local/bin/gfortran in your case), and /path/to/gcc with the output of `which g++`. Something like: `FC=/usr/local/bin/gfortran CXX=/usr/local/bin/g++ cmake ..`

**Reply by jas220** [reply]
Here's what I get.
<img width="877" alt="Screen Shot 2021-04-20 at 9 54 42 PM" src="https://user-images.githubusercontent.com/82110485/115463338-9c9caa00-a223-11eb-847a-beef500fb893.png">

**Reply by fonsocarre** [reply]
Then I'm afraid your gfortran installation is broken. You probably need to cleanup and reinstall it. Try again once that's done by deleting the `build` folder, creating it again and running the `FC=/usr/local/bin/gfortran CXX=/usr/bin/g++ cmake ..` command.

**Reply by jas220** [reply]
Ok, thank you, that worked!
Now, where do I get the SHARPy variables?

![Uploading Screen Shot 2021-04-20 at 10.16.15 PM.png…]()

**Reply by jas220** [reply]
<img width="680" alt="Screen Shot 2021-04-20 at 10 16 15 PM" src="https://user-images.githubusercontent.com/82110485/115466054-6e20ce00-a227-11eb-8c94-4d3bb9006a47.png">

**Reply by ngoiz** [reply]
You are not looking for the correct directory. When you do `source <path>` that path is relative. In your case you are in the `build` directory hence you need to go up a directory to find `bin` (`source ../bin/sharpy_vars.sh`). Using the TAB key to autocomplete the available options at each level is always helpful
